profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrActivator.java

```diff
@@ -83,35 +81,42 @@ private boolean notClearForTakeoff(ConfigProperties config) {
       return true;
     }
 
-    Path outputDir = Paths.get(config.getString(CONFIG_KEY_PROFILER_DIRECTORY));
+    return false;
+  }
+
+  private boolean checkOutputDir(Path outputDir) {
     if (!Files.exists(outputDir)) {
       // Try creating the directory for the user...
       try {
         Files.createDirectories(outputDir);
       } catch (IOException e) {
-        return outdirWarn(outputDir, "does not exist and could not be created");
+        outdirWarn(outputDir, "does not exist and could not be created");
+        return false;
       }
     }
     if (!Files.isDirectory(outputDir)) {
-      return outdirWarn(outputDir, "exists but is not a directory");
+      outdirWarn(outputDir, "exists but is not a directory");
+      return false;
     }
 
     if (!Files.isWritable(outputDir)) {
-      return outdirWarn(outputDir, "exists but is not writable.");
+      outdirWarn(outputDir, "exists but is not writable");
+      return false;
     }
-    return false;
-  }
 
-  private boolean outdirWarn(Path dir, String suffix) {
-    logger.log(
-        WARNING,
-        "PROFILER WILL NOT BE STARTED: The configured output directory {0} {1}.",
-        new Object[] {dir, suffix});
     return true;
   }
 
+  private void outdirWarn(Path dir, String suffix) {
+    logger.log(WARNING, "The configured output directory {0} {1}.", new Object[] {dir, suffix});
+  }
+
   private void activateJfrAndRunForever(ConfigProperties config, Resource resource) {
+    boolean keepFiles = keepFiles(config);
     Path outputDir = Paths.get(config.getString(CONFIG_KEY_PROFILER_DIRECTORY));
+    if (keepFiles && !checkOutputDir(outputDir)) {
+      keepFiles = false;
```

Comment:
Ah ok, I think this answers the question I asked above about the behavior. I think the user should be told clearly here that files will not be written.

Reply:
The warning about output directory not being writable should be enough to inform user why there are not recordings. Also consider that the `keepfiles` flag is only useful for debugging the agent, from the users perspective it does not make sense to use the `keepfiles` flag.